examples/case_studies/reliability_and_calibrated_prediction.myst.md

```diff
@@ -0,0 +1,1191 @@
+    betas = joint_draws["beta"].values
+    for i in range(len(joint_draws)):
+        fit = weibull_min(c=alphas[i], scale=betas[i])
+        rho = fit.cdf(up) - fit.cdf(lp) / (1 - fit.cdf(lp))
+        lb = binom(n_at_risk, rho).ppf(0.05)
+        ub = binom(n_at_risk, rho).ppf(0.95)
+        point_prediction = n_at_risk * rho
+        records.append([alphas[i], betas[i], rho, n_at_risk, lb, ub, point_prediction])
+    return pd.DataFrame(
+        records, columns=["alpha", "beta", "rho", "n_at_risk", "lb", "ub", "expected"]
+    )
+
+
+output_df = PI_failures(joint_draws, 150, 600, 1700)
+output_df
+```
+
+```{code-cell} ipython3
+def cost_func(failures, power):
+    ### Imagined cost function for failing item e.g. refunds required
+    return -np.power(failures, power)
+
+
+mosaic = """AAAA
+            BBCC"""
+fig, axs = plt.subplot_mosaic(mosaic=mosaic, figsize=(20, 12))
+axs = [axs[k] for k in axs.keys()]
+ax = axs[0]
+ax1 = axs[1]
+ax2 = axs[2]
+
+ax.scatter(
+    output_df["alpha"],
+    output_df["expected"],
+    c=output_df["beta"],
+    cmap=cm.cool,
+    alpha=0.3,
+    label="Coloured by function of Beta values",
+)
+ax.legend()
+ax.set_ylabel("Expected Failures")
+ax.set_xlabel("Alpha")
+ax.set_title(
+    "Posterior Predictive Expected Failure Count between 150-600 hours \nas a function of Weibull(alpha, beta)",
+    fontsize=20,
+)
+
+ax1.hist(
+    output_df["lb"],
+    ec="black",
+    color="slateblue",
+    label="95% PI Lower Bound on Failure Count",
+    alpha=0.3,
+)
+ax1.axvline(output_df["lb"].mean(), label="Expected 95% PI Lower Bound on Failure Count")
+ax1.axvline(output_df["ub"].mean(), label="Expected 95% PI Upper Bound on Failure Count")
+ax1.hist(
+    output_df["ub"],
+    ec="black",
+    color="cyan",
+    label="95% PI Upper Bound on Failure Count",
+    bins=20,
+    alpha=0.3,
+)
+ax1.hist(
+    output_df["expected"], ec="black", color="pink", label="Expected Count of Failures", bins=20
+)
+ax1.set_title("Uncertainty in the Posterior Prediction Interval of Failure Counts", fontsize=20)
+ax1.legend()
+
+ax2.set_title("Expected Costs Distribution(s)  \nbased on implied Failure counts", fontsize=20)
+ax2.hist(
+    cost_func(output_df["expected"], 2.3),
+    label="Cost(failures,2)",
+    color="royalblue",
+    alpha=0.3,
+    ec="black",
+    bins=20,
+)
+ax2.hist(
+    cost_func(output_df["expected"], 2),
+    label="Cost(failures,2.3)",
+    color="red",
+    alpha=0.5,
+    ec="black",
+    bins=20,
+)
+ax2.set_xlabel("$ cost")
+ax2.legend()
+```
+
+The choice of model in such cases is crucial. The decision about which failure profile is apt has to be informed by a subject matter expert because extrapolation from such sparse data is always risky. An understanding of the uncertainty is crucial if real costs attach to the failures and the subject matter expert is usually better placed to tell if you 2 or 7 failures can be plausibly expected within 600 hours of service. 
+
+# Conclusion
```

Comment:
I think this should be level 2 header, otherwise the table of contents looks a bit weird: https://pymcio--491.org.readthedocs.build/projects/examples/en/491/case_studies/reliability_and_calibrated_prediction.html.

Reply:
Done, changed this.